### ISSUE TITLE
Disabled query is run when useQuery is nested within a usePrefetchQuery

### ISSUE THREAD

**Issue by Roystbeef** [TanStack/query#7982]
### Describe the bug

```jsx
import React from "react";
import { usePrefetchQuery, useQuery } from "@tanstack/react-query";

function getTimeoutQuery() {
  return {
    queryKey: ["timeoutQuery"],
    queryFn: () =>
      new Promise((resolve) => {
        console.log("we should not be seeing this log");
        setTimeout(() => resolve("done"), 3000);
      }),
    enabled: false,
  };
}

export default function Home() {
  usePrefetchQuery(getTimeoutQuery());

  return <HomeInner />;
}

function HomeInner() {
  const timeoutQuery = useQuery(getTimeoutQuery());

  return <>{timeoutQuery.status}</>;
}
```
when using both `usePrefetchQuery` and `useQuery`, the query function is ran even though both queries have `enabled` set to false.

### Your minimal, reproducible example

https://glitch.com/edit/#!/mighty-ebony-skink

### Steps to reproduce

1. Go to the glitch.me or copy/paste the code shared above
2. Run the react app

Expected Behavior:
- `timeoutQuery.status` is `idle` and the `queryFn` is not run

Actual Behavior:
- `timeoutQuery.status` goes from `pending` -> `success` and the `queryFn` is run even though `enabled` is set to `false`

### Expected behavior

As a user, I expected `timeoutQuery.status` to always be `idle` since the query is not enabled

### How often does this bug happen?

Every time

### Platform

- OS: MacOS
- Browser: Arc (also saw this in jest)
- Version: 1.57.2 (53036) / Chromium Engine Version 128.0.6613.85

### Tanstack Query adapter

None

### TanStack Query version

v5.52.2

### Additional context

Totally recognize I could be misunderstanding `usePrefetchQuery` - but I was not able to find details to build a mental model of how this should work in the docs for `usePrefetchQuery` or for using the `enabled` flag

**Comment by TkDodo** [comment]
`usePrefetchQuery` doesn't allow for an `enabled` field. It's for unconditional prefetching of queries.

**Comment by k-ruben** [comment]
I encountered the same problem. I would like to disable the prefetchQuery until another query returned an Id that the prefetchQuery is depending on. It seems logical to me that I should be able to disable the prefetchQuery until then.

Now I changed the prefetchQuery to a normal query which is fine I guess, but I like to convey the intend of the query in my code.

In my opinion the documentation should contain this information: https://tanstack.com/query/latest/docs/framework/react/reference/usePrefetchQuery
